examples/app.js

```diff
@@ -1,71 +1,98 @@
+        nonce: 42,
+        value: "1",
+        receiver: new Address("erd1uv40ahysflse896x4ktnh6ecx43u7cmy9wnxnvcyp7deg299a4sq6vaywa"),
+        gasPrice: 1000000000,
+        gasLimit: 50000,
+        data: "",
+        chainID: "T",
+        version: 1
     });
 
-    // Sign a transaction
-    let firstTransaction = await provider.signTransaction({
-        toPlainObject: function() {
-            return {
-                nonce: 42,
-                value: "1",
-                receiver: "erd1uv40ahysflse896x4ktnh6ecx43u7cmy9wnxnvcyp7deg299a4sq6vaywa",
-                gasPrice: 1000000000,
-                gasLimit: 50000,
-                data: "",
-                chainID: "T"
-            };
-        }
-    });
+    let firstTransactionSigned = await provider.signTransaction(firstTransaction);
 
+    assert(firstTransaction === firstTransactionSigned, "The extension provider should return the same object passed as input");
     console.log("First transaction, upon signing:");
     console.log(firstTransaction);
+}
 
-    // Sign & broadcast another transaction.
-    // This should fail (bad nonce etc.)
-    await provider.sendTransaction({
-        toPlainObject: function() {
-            return {
-                nonce: 43,
-                value: "1",
-                receiver: "erd1uv40ahysflse896x4ktnh6ecx43u7cmy9wnxnvcyp7deg299a4sq6vaywa",
-                gasPrice: 1000000000,
-                gasLimit: 200000,
-                data: Buffer.from("hello").toString("base64"),
-                chainID: "T"
-            };
-        }
-    });
+export async function signMessages() {
+    let provider = ExtensionProvider.getInstance();
 
-    // Sign a message.
-    let message = await provider.signMessage({
-        message: "hello"
+    let message = new DummyMessage({
+        message: Buffer.from("hello")
     });
+    let messageSigned = await provider.signMessage(message);
 
+    assert(message === messageSigned, "The extension provider should return the same object passed as input");
     console.log("Message, upon signing:");
-    console.log(message);
+    console.log(messageSigned);
+}
+
+class DummyTransaction {
+    nonce = 0;
+    value = "";
+    receiver = null;
+    sender = null;
+    gasPrice = 0;
+    gasLimit = 0;
+    data = "";
+    chainID = "";
+    version = 0;
+    options = undefined;
+    signature = null;
+
+    constructor(init) {
+        Object.assign(this, init);
+    }
+
+    toPlainObject() {
+        return {
+            nonce: this.nonce,
+            value: this.value,
+            receiver: this.receiver.bech32(),
+            gasPrice: this.gasPrice,
+            gasLimit: this.gasLimit,
+            data: this.data,
+            chainID: this.chainID,
+            version: this.version,
+            options: this.options
+        };
+    }
+
+    applySignature(signature, signedBy) {
+        console.log("applySignature()", signature.hex(), signedBy.bech32());
```

Comment:
nu cred ca vrem sa lasam console.log-uri prin fisiere in prod

Reply:
This is in the `examples` (not delivered in the npm package) - is it OK to keep them?